waku/waku_rln_relay/rln_relay.nim

```diff
@@ -308,21 +300,12 @@ proc generateRlnValidator*(wakuRlnRelay: WakuRLNRelay,
         info "message bandwidth limit exceeded, running rate limit proof validation"
     except OverflowDefect: # not a problem
       debug "not enough bandwidth, running rate limit proof validation"
-   
 
     let decodeRes = WakuMessage.decode(message.data)
     if decodeRes.isOk():
-      let
-        wakumessage = decodeRes.value
-        payload = string.fromBytes(wakumessage.payload)
-
-      # check the contentTopic
-      if (wakumessage.contentTopic != "") and (contentTopic != "") and (wakumessage.contentTopic != contentTopic):
-        trace "content topic did not match:", contentTopic=wakumessage.contentTopic, payload=payload
-        return pubsub.ValidationResult.Accept
-
```

Comment:
how should we prevent replay attacks on different topics if we do not perform this check?

Reply:
You mean using a message (with a given content topic) + proof created by someone else to spam other pubsub topics?

Mind ellaborating also on this part? I don't fully understand why its Accepting the message when !=

Reply:
yeah, so imagine that I craft a message for a given content-topic/pubsub-topic pair, and someone takes that message and sends it on a different content-topic/pubsub-topic pair

now with sharding, this may not be an issue, but it will still be possible to send messages from one shard onto another (where it doesn't belong)

Reply:
With vanila relay (just for status by now), the idea was to use the timestamps to prevent this, so we only accept a window of x around the timestamp. But well, timestamp is currently not mandatory.

https://github.com/waku-org/nwaku/blob/b3bb7a1113c6d584e30eecfd724175461f0ada9c/apps/wakunode2/wakunode2_validator_signed.nim#L61

But I'm missing something in here.

Reply:
we can just use the epoch for this, its mandatory!

Reply:
I think that would solve the problem